src/Commands/Workflow/Info/InfoBaseCommand.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Workflow\Info;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+
+abstract class InfoBaseCommand extends TerminusCommand implements SiteAwareInterface
+{
+    use SiteAwareTrait;
+
+    /**
+     * Get the Workflow object.
+     */
+    protected function getWorkflow($site_id, $options)
+    {
+        if (!($options['workflow-id'] || $options['latest-with-logs'])) {
+            $this->log()->error('Required option missing: --workflow-id or --latest-with-logs.');
+            return;
+        }
+        $site = $this->getSite($site_id);
+
+        if ($options['workflow-id']) {
+            $workflow_id = $options['workflow-id'];
+            $model_data  = (object)['id' => $workflow_id];
+            $workflow    = $site->workflows->add($model_data);
+        } elseif ($options['latest-with-logs']) {
+            $site->workflows->fetch(['paged' => false]);
+            $workflow = $site->workflows->findLatestWithLogs();
+            if (!$workflow) {
+                $this->log()->info('No recent workflow has logs');
+                return;
+            }
+        }
+        try {
+            $workflow->fetchWithLogs();
```

Comment:
It looks like _all_ workflows are being fetched with logs. Either:
- cases using `workflow-id` shouldn't have logs fetched
- `-with-logs` should be omitted
- `--with-logs` should be separated from `--latest` and can be paired with the workflow ID